provider/vsphere/environ_availzones.go

```diff
@@ -111,15 +116,27 @@ func (env *environ) parseAvailabilityZones(args environs.StartInstanceParams) ([
 			return nil, errors.Trace(err)
 		}
 	}
-	zoneInstances, err := AvailabilityZoneAllocations(env, group)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	logger.Infof("found %d zones: %v", len(zoneInstances), zoneInstances)
-
 	var zoneNames []string
-	for _, z := range zoneInstances {
-		zoneNames = append(zoneNames, z.ZoneName)
+	// Vsphere will misbehave if we call AvailabilityZoneAllocations with empty
+	// groups, in this case all zones shouhld be returned.
+	if len(group) != 0 {
+		zoneInstances, err := AvailabilityZoneAllocations(env, group)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		for _, z := range zoneInstances {
+			zoneNames = append(zoneNames, z.ZoneName)
+		}
+		logger.Infof("found %d zones: %v", len(zoneInstances), zoneInstances)
+	} else {
+		zoneInstances, err := AllAvailabilityZones(env)
```

Comment:
It looks like this function returns `Zones` rather than `ZoneInstance`s - it'd be clearer to name the variable `zones`. (I was going to suggest that you move the name-collection outside the loop until I realised they were different types.)